### PR TITLE
add starship requirements in dockerfile, add docker gh action

### LINKED ISSUE #53: Publish Docker Images on new releases
@Anmol1696 is using them for testing in the Starship environment, and others may find them useful as well.

### PR DESCRIPTION
**Pull request by Anmol1696**
## Overview
Add starship requirements into the dockerfile
Add gh-action to push docker images to ghcr.io at:
```bash
ghcr.io/osmosis-labs/meshd
```

Closes #53

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*.*.*"
```
Comment (alpe) [inline .github/workflows/docker.yaml:8]: Just checking, does this covers RC tags like `v0.1.2-rc.3` as well?

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  workflow_dispatch:
+    inputs:
+      tags:
+        description: "Tags to build and push"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+  packages: write
```
Comment (alpe) [inline .github/workflows/docker.yaml:21]: We need this to push to the docker registry, right?

File: .github/workflows/docker.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        with:
+          go-version: "1.20"
+          check-latest: true
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.DOCKER_REGISTRY }}/${{ env.DOCKER_IMAGE }}
+          tags: |
```
Comment (alpe) [inline .github/workflows/docker.yaml:48]: Do you use the tags for anything?

**Review by alpe — APPROVED** [review]
LGTM! 
Just some questions to better understand the changes

File: .github/workflows/docker.yaml (hunk 2 above)
Comment (Anmol1696) [reply .github/workflows/docker.yaml:21]: yup

File: .github/workflows/docker.yaml (hunk 3 above)
Comment (Anmol1696) [reply .github/workflows/docker.yaml:48]: for versioning. As more versions of the mesh-security-sdk comes, we should be able to use the tagged version

File: .github/workflows/docker.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run go vendor
+        run: |
+          cd demo/ && go mod vendor
```
[the remark this answers is not in the document]
Comment (Anmol1696) [reply .github/workflows/docker.yaml:70]: Yes, useally i would just have this part of the dockerfile itself, but the current system seemed to take pre built vendor dir

File: demo/Dockerfile [hunk 5]
```diff
@@ -31,9 +30,15 @@ FROM alpine:3.16
 
 COPY --from=go-builder /code/build/meshd /usr/bin/meshd
 
+# Set up dependencies used for Starship
+ENV PACKAGES curl make bash jq sed
```
[the remark this answers is not in the document]
Comment (Anmol1696) [reply demo/Dockerfile:34]: Can do. will change it.

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (Anmol1696) [reply .github/workflows/docker.yaml:8]: I just took it from: https://github.com/osmosis-labs/mesh-security-sdk/blob/main/.github/workflows/release.yml#L6
I would like to have something like: 
https://github.com/cosmos/cosmos-sdk/blob/main/.github/workflows/docker.yml#L11...L13

But did not want to deviate from the repo practices

**Pull request merged**